modules/backend/formwidgets/codeeditor/assets/less/codeeditor.less

```diff
@@ -1,103 +1,665 @@
+
     .editor-toolbar {
-        position: absolute;
-        padding: 0 5px;
-        bottom: 10px;
-        right: 25px;
+        display: flex;
+        flex-direction: row;
+        align-items: center;
+        gap: 20px;
+        height: 24px;
+        font-size: 11px;
+        position: relative;
+        flex-grow: 0;
+        flex-shrink: 0;
+        padding: 2px 8px;
         z-index: @zindex-form;
         background: rgba(0,0,0,.8);
-        border-radius: 5px;
 
-        > ul, ul > li { list-style-type: none; padding: 0; margin: 0; }
-        > ul > li {
-            float: left;
+        &::before {
+            content: '';
+            position: absolute;
+            top: 0;
+            left: 0;
+            width: 100%;
+            height: 100%;
+            z-index: 9;
+            background: rgba(0, 0, 0, 0.08);
+        }
 
-            .tooltip.left {
-                margin-right: 25px;
-            }
+        &.is-dark::before {
+            background: rgba(255, 255, 255, 0.08);
         }
-        > ul > li > a {
-            display: block;
-            height: 25px;
-            width: 25px;
-            color: #666;
-            font-size: 20px;
-            text-align: center;
-            text-decoration: none;
-            text-shadow: 0 0 5px #000;
-
-            > abbr {
-                position: absolute;
-                .text-hide();
-            }
-            > i {
-                .opacity(1);
-                display: block;
-                &:before {
-                    font-size: 15px;
+
+        > div {
+            z-index: 11;
+        }
+
+        .position {
+            flex-grow: 1;
+        }
+
+        .actions {
+            .action {
+                margin: -2px 0;
+                padding: 0 5px;
+                font-size: 1.1rem;
+                color: inherit;
+                opacity: 0.4;
+
+                &:hover {
+                    opacity: 0.8;
                 }
-            }
-            &:hover, &:focus {
-                > i {
-                    .opacity(1);
-                    color: #fff;
+
+                &.active {
+                    opacity: 1;
                 }
             }
         }
     }
+}
 
-    &.editor-fullscreen {
-        z-index: @zindex-fullscreen + 1;
-        position: fixed !important;
-        top: 0;
-        left: 0;
-        height: 100%;
-        border-width: 0;
-        .border-radius(0);
-        .editor-toolbar {
-            z-index: @zindex-fullscreen + 2;
-        }
-        .editor-code { .border-radius(0); }
-        .ace_search { z-index: @zindex-fullscreen + 3; }
-    }
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
```

Comment:
Can we import this from a separate file?

Reply:
I can have another look, but from what I remember, the CSS is inlined in Monaco so it may not be importable.